react/react1/week3/todo-app/src/contexts/ToDosContext.js

```diff
@@ -0,0 +1,73 @@
+import { createContext, useEffect, useReducer } from 'react';
+
+export const ToDosContext = createContext();
+
+const initialState = {
+  todos: [],
+};
+
+const toDosReducer = (state, action) => {
+  switch (action.type) {
+    case 'GET_TODOS':
+      return { todos: action.payload };
+    case 'ADD_TODO':
+      return { todos: [...state.todos, action.payload] };
+    case 'UPDATE_TODO':
+      const updatedTodos = state.todos.map((todo) =>
+        todo.id === action.payload.id ? action.payload : todo
+      );
+      return { todos: updatedTodos };
+    default:
+      return state;
+  }
+};
+
+export const ToDosProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(toDosReducer, initialState);
+
+  useEffect(() => {
+    async function fetchData() {
+      try {
+        const response = await fetch(
+          'https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw'
+        );
+        const data = await response.json();
+        if (data) {
```

Comment:
If I am not mistaken, the await here takes care of the data, so the conditional is most probably not necessary 🤔

Reply:
I think I wanted to check if the response was ok.
I modified it this way:
if (response.ok) {
        const data = await response.json();
        dispatch({ type: 'GET_TODOS', payload: data });
      }